### DISCUSSION TITLE
Buena noche compañeros explorers, solicito ayuda con git.

### DISCUSSION THREAD

**Discussion by LJavier33** [Launch-X-Latam/MisionFrontEnd#585] [Team Posts]
Buena noche compañeros, reciban un  saludo, espero me puedan asesorar en un inconveniente, la verdad soy nuevo en git y git hub. Subí de manera errónea unas practicas de frontend a mis repositorios, cuando vi que no habia sido de la forma correcta elimine los mismos y cree otros pero al momento de querer subir de nuevo mis practicas no me es posible y ya no he podido revertir mi error.
Primero me salia esto:
" error: remote fatal origin already exist". 
Borre el "origin" que ya habia generado con un "git remote on remove" pero despues 
Cuando  ejecuto el "git status" para validar mis archivos de la carpeta y de nuevo subir mi proyecto y mis archivos me sale esto:
"nothing to commit, working tree clean".
He buscado la solución en internet y en varios  sitios pero no me queda claro y no entiendo que tengo que hacer.
Espero me puedan echar la mano con algún tip o un comando o como lo puedo solucionar, ya casi acaba esto y me siento en el calle de la amargura. Saludos.

**Comment by Amhedriel** [comment]
## Buenas colega

Que te sale cuando haces el merge¿?

Y que es lo que quieres hacer, recuperar tus antiguos archivos? o enviar tus avances a git, porque hasta donde leo, creo que tienes que borrar los repositorios erróneos en GitHub y luego en Git, porque donde dice: "nothing to commit, working tree clean" es porque no estas haciendo un merge antes del commit, tienes que hacer un `git add archivoParaSubie.md` o `git add .` pero este último considera una mala práctica, ya que sube todos los archivos que tengas en esa carpeta, excepto tu `.gitignore`.

Luego `git commit -m "mensaje breve para el historial"`

Si necesitas agregar otro archivo y no has hecho el `push` utiliza `git commit -m "" --amend`

Eh intenta `git push origin main/master` la rama en la que estás haciendo el `add` a veces es `main` a veces es `master`

Sin embargo, si borraste todo tienes que volver a usar `git init` para subir el repositorio nuevo y sigue las instrucciones de GitHub.

Si no borraste tu repositorio intenta actualizarlo al `branch` local: `git fetch` para bajar un índice de los cabios que hayas hecho.

`git pull origin main/master` para bajar los cambios a nuestro `brnach` local haciendo un **automerge** con nuestros cambios.

Si tenemos cambios sin *commitear*, Git nos pedirá que hagamos `commit` (en local), **no hacer ``push``** antes de hacer ``pull``.

Y si esto tampoco te soluciona nada... revisa mi [tutorial](https://github.com/Amhedriel/Tutoriales_Git_BASH_MarkDown "Repositorio de Git") o directamente revisa la [documentación oficial](https://git-scm.com/book/es/v2 "Git book en español"), saludos.

**Comment by Standard-IO** [comment]
Como tal no estas describiendo el error. No te sugiero seguir la respuesta de @Amhedriel parece que tiene unos problemas con los términos que se manejan en git y puede confundirte mucho más. 

Creo que el comando que has puesto acá no del todo correcto `git remote on remove` , ya que no existe el comando `on` para operaciones **remote**. Pero todo parece indicar que eliminaste el la unión con el repositorio remoto *github*.
Puedes ver si esto es correcto ejecutando `git remote --verbose` para ver si tu repositorio mantiene seguimiento del repositorio en *github*, deberíá de salir una lista con los repositorios a los que esta ligado este repositorio (entre ellos el de *github*, si es que no lo borraste) y nada si lo borraste. En ese caso solo debes de volver a agregarlo como repositorio remoto.

El hecho de que te aparezca `nothing to commit, working tree clean` solo quiere decir que no tienes documentos con cambios y puestos en el **stage area** (área donde se ponen los archivos agregados con el comando `git add`  o ` git rm` antes de ser aplicados con un `commit`)  por lo tanto no hay cambios que aplicar, además, si borraste el seguimiento del repositorio remoto no te dará el seguimiento entre las diferencias entre tu repositorio local y remoto (porque tu repositorio solo esta en modo local). En general `nothing to commit, working tree clean` aparece cuando todo esta bien, el único caso que no esta bien es cuando estas siguiendo un repositorio remoto y no se ven reflejados las diferencias entre los commits.

En conclusión solo debes de volver a agregar el repositorio remoto. 

Nota: para remover archivos en git se realiza con dos comandos porque el que pusiste es para eliminar el emparejamiento con repositorios remotos. Para eliminar un archivo de un repositorio primero se elimina su seguimiento en git con `git rm <file name>` (para eliminarlo del **index** de git, que es el que lleva registro de que archivos debe de monitoriar el repositorio) y después se borra el archivo con `rm`, linux, o sea ( borrarlo de la forma normal que haces con el ratón de toda la vida).

**Comment by shelkanife** [comment]
Hola, no sé si aún tienes el problema pero puedes solucionarlo de dos maneras:
- borrando la carpeta .git (Si no quieres perder el historial de tus commits no uses esta opción)
    - crea tu nuevo repo e inicializa de nueva cuenta tu repo local  y añade el nuevo remote
 
- cambiando el origin de tu repo
	- <code>git remote set-url origin <liga a tu nuevo repo\></code>
	- <code>git remote -v </code> (verifica que el origin haya cambiado)

**Comment by LJavier33** [comment]
Gracias colegas, por el asesoramiento, realizaré lo que me comentan y espero poder resolverlo al a brevedad. Saludos.

**Comment by Standard-IO** [comment]
La opción de @shelkanife me parece buena. Es fácil de seguir y probablemente de esta manera no tengas problemas. Deberías de comenzar por ahí y no perderás mucho tiempo.

**Comment by LJavier33** [comment]
@Amhedriel  En este caso, quisiera borrar todo lo que hice al inicio, hacer de cuenta que no paso XD y empezar desde cero y subir mis proyectos como si fuera la primera vez.

**Comment by LJavier33** [comment]
@shelkanife   Vale vale, lo hare de esa manera, gracias colega.

**Comment by LJavier33** [comment]
@Isra-ResVo  Haaaa ok ok. gracias por la sugerencia, asi lo hare, gracias por el comentario.

**Comment by LJavier33** [comment]
@shelkanife  Hola colega,  disculpa ¿Cómo borro la carpeta git, es decir hago esto mediante algún comando o borro la carpeta ingresando a mi disco C:/ donde esta alojada?

**Comment by Standard-IO** [comment]
@LJavier33  solo borra la carpeta como normalmente harías con cualquier carpeta. Git es simple, si borras la carpeta borras el repositorio, en esa carpteta se encuentra toda la información de las ediciones, configuraciones locales y nada más. Por lo que si la borras no afecta los archivos fuera de ella.  Después,  puedes iniciar uno nuevo en el mismo proyecto (carpeta padre) sin mayor dificultad con  el comando`git init`. 

Si estas en Windows, que parece que sí, solo selecciona la carpeta `.git` dentro del proyecto,  CLICK derecho y borrar.

**Comment by shelkanife** [comment]
@LJavier33 Pues, entra a tu repositorio local (carpeta de tu proyecto) y ahí estará. Es una carpeta oculta, por lo que, si lo haces desde el explorador de archivo de Windows habilita esa opción y después sigue lo que dice @Isra-ResVo. Si estás desde un editor de código o un IDE selecciona la carpeta y bórrala.

**Comment by LJavier33** [comment]
@shelkanife , @Isra-ResVo   Es verdad, así se resolvió, muchas, muchisimas gracias por los tips y la información. Estoy verde, pero preparándome más y con la ayuda de colegas como ustedes y los Sherpas, se que la puedo armar, gracias de nuevo.

**Comment by LJavier33** [comment]
@Amhedriel  Gracias colega, también por tu aportación a mi duda, saludos.

**Comment by LJavier33** [comment]
Saludos compañeros, ¿Disculpen les ha pasado que suben su proyecto y marca todo bien en el Git pero al momento de validar en el Github no se carga nada, solo se ve el archivo README y ya, hay algún detalle por el que pasa esto? Ya subí un proyecto a un repositorio y todo bien, pero hago otro repositorio y me pasa eso.

**Comment by Standard-IO** [comment]
No estas siguiendo correctamente las instrucciones para crear el repositorio. Al momento de crear el repositorio no debes de tener la opción `Add a README file` seleccionada. Debe ser un repositorio vacío. De ahí nadams lo importas y ya.

<br/>
incluso al crear el repositorio en las opciones dice que no debes de hacerlo si lo que quieres es importar un repositorio local.
<img src="https://user-images.githubusercontent.com/70794050/207993234-7a163067-23ce-445a-8fa0-77d70eadc79e.png"  height=250px/>

E

**Comment by LJavier33** [comment]
Correcto @Isra-ResVo   tea gradezco por el dato.
